// @pact-foundation/pact/src/v4/http/requestBuilder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestBuilder = void 0;
var ramda_1 = require("ramda");
var matchers_1 = require("../../v3/matchers");
var _1 = require(".");
var RequestBuilder = /** @class */ (function () {
    // tslint:disable:no-empty-function
    function RequestBuilder(interaction) {
        this.interaction = interaction;
    }
    RequestBuilder.prototype.query = function (query) {
        var _this = this;
        (0, ramda_1.forEachObjIndexed)(function (v, k) {
            if (Array.isArray(v)) {
                v.forEach(function (vv, i) {
                    _this.interaction.withQuery(k, i, (0, matchers_1.matcherValueOrString)(vv));
                });
            }
            else {
                _this.interaction.withQuery(k, 0, (0, matchers_1.matcherValueOrString)(v));
            }
        }, query);
        return this;
    };
    RequestBuilder.prototype.headers = function (headers) {
        var _this = this;
        (0, ramda_1.forEachObjIndexed)(function (v, k) {
            if (Array.isArray(v)) {
                v.forEach(function (header, index) {
                    _this.interaction.withRequestHeader("".concat(k), index, (0, matchers_1.matcherValueOrString)(header));
                });
            }
            else {
                _this.interaction.withRequestHeader("".concat(k), 0, (0, matchers_1.matcherValueOrString)(v));
            }
        }, headers);
        return this;
    };
    RequestBuilder.prototype.jsonBody = function (body) {
        this.interaction.withRequestBody((0, matchers_1.matcherValueOrString)(body), 'application/json');
        return this;
    };
    RequestBuilder.prototype.binaryFile = function (contentType, file) {
        var body = (0, _1.readBinaryData)(file);
        this.interaction.withRequestBinaryBody(body, contentType);
        return this;
    };
    RequestBuilder.prototype.multipartBody = function (contentType, file, mimePartName) {
        this.interaction.withRequestMultipartBody(contentType, file, mimePartName);
        return this;
    };
    RequestBuilder.prototype.body = function (contentType, body) {
        this.interaction.withRequestBinaryBody(body, contentType);
        return this;
    };
    return RequestBuilder;
}());
exports.RequestBuilder = RequestBuilder;
