// @pact-foundation/pact/src/v4/http/requestWithPluginBuilder.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RequestWithPluginBuilder = void 0;
var requestBuilder_1 = require("./requestBuilder");
var RequestWithPluginBuilder = /** @class */ (function (_super) {
    __extends(RequestWithPluginBuilder, _super);
    function RequestWithPluginBuilder() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RequestWithPluginBuilder.prototype.pluginContents = function (contentType, contents) {
        this.interaction.withPluginRequestInteractionContents(contentType, contents);
        return this;
    };
    RequestWithPluginBuilder.prototype.query = function (query) {
        _super.prototype.query.call(this, query);
        return this;
    };
    RequestWithPluginBuilder.prototype.headers = function (headers) {
        _super.prototype.headers.call(this, headers);
        return this;
    };
    RequestWithPluginBuilder.prototype.jsonBody = function (body) {
        _super.prototype.jsonBody.call(this, body);
        return this;
    };
    RequestWithPluginBuilder.prototype.binaryFile = function (contentType, file) {
        _super.prototype.binaryFile.call(this, contentType, file);
        return this;
    };
    RequestWithPluginBuilder.prototype.multipartBody = function (contentType, file, mimePartName) {
        _super.prototype.multipartBody.call(this, contentType, file, mimePartName);
        return this;
    };
    RequestWithPluginBuilder.prototype.body = function (contentType, body) {
        _super.prototype.body.call(this, contentType, body);
        return this;
    };
    return RequestWithPluginBuilder;
}(requestBuilder_1.RequestBuilder));
exports.RequestWithPluginBuilder = RequestWithPluginBuilder;
